src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,342 @@
+      conv_lhs { rw ← set.inter_union_compl i T },
+      rw [measure_union, show j₂.μ (i ∩ T) = 0, by exact nonpos_iff_eq_zero.1
+            (hT₄ ▸ measure_mono (set.inter_subset_right _ _)), zero_add],
+      { exact set.disjoint_of_subset_left (set.inter_subset_right _ _)
+          (set.disjoint_of_subset_right (set.inter_subset_right _ _) disjoint_compl_right) },
+      { exact hi.inter hT₁ },
+      { exact hi.inter hT₁.compl } },
+    rw [← ennreal.to_real_eq_to_real (measure_lt_top _ _) (measure_lt_top _ _),
+        hμ₁, hμ₂, ← hj],
+    exact of_inter_eq_of_symm_diff_eq_zero_positive hS₁.compl hT₁.compl hi hS₃ hT₃ hST₁,
+    all_goals { apply_instance } },
+  { refine measure_theory.measure.ext (λ i hi, _),
+    have hν₁ : (j₁.ν i).to_real = - j₁.to_signed_measure (i ∩ S),
+    { rw [to_signed_measure, sub_to_signed_measure_apply (hi.inter hS₁),
+          show j₁.μ (i ∩ S) = 0, by exact nonpos_iff_eq_zero.1
+            (hS₄ ▸ measure_mono (set.inter_subset_right _ _)),
+          ennreal.zero_to_real, zero_sub],
+      conv_lhs { rw ← set.inter_union_compl i S },
+      rw [measure_union, show j₁.ν (i ∩ Sᶜ) = 0, by exact nonpos_iff_eq_zero.1
+            (hS₅ ▸ measure_mono (set.inter_subset_right _ _)), add_zero, neg_neg],
+      { exact set.disjoint_of_subset_left (set.inter_subset_right _ _)
+          (set.disjoint_of_subset_right (set.inter_subset_right _ _) disjoint_compl_right) },
+      { exact hi.inter hS₁ },
+      { exact hi.inter hS₁.compl } },
+    have hν₂ : (j₂.ν i).to_real = - j₂.to_signed_measure (i ∩ T),
+    { rw [to_signed_measure, sub_to_signed_measure_apply (hi.inter hT₁),
+          show j₂.μ (i ∩ T) = 0, by exact nonpos_iff_eq_zero.1
+            (hT₄ ▸ measure_mono (set.inter_subset_right _ _)),
+          ennreal.zero_to_real, zero_sub],
+      conv_lhs { rw ← set.inter_union_compl i T },
+      rw [measure_union, show j₂.ν (i ∩ Tᶜ) = 0, by exact nonpos_iff_eq_zero.1
+            (hT₅ ▸ measure_mono (set.inter_subset_right _ _)), add_zero, neg_neg],
+      { exact set.disjoint_of_subset_left (set.inter_subset_right _ _)
+          (set.disjoint_of_subset_right (set.inter_subset_right _ _) disjoint_compl_right) },
+      { exact hi.inter hT₁ },
+      { exact hi.inter hT₁.compl } },
+    rw [← ennreal.to_real_eq_to_real (measure_lt_top _ _) (measure_lt_top _ _),
+        hν₁, hν₂, neg_eq_iff_neg_eq, neg_neg, ← hj],
+    exact eq.symm (of_inter_eq_of_symm_diff_eq_zero_negative hS₁ hT₁ hi hS₂ hT₂ hST₂),
+    all_goals { apply_instance } }
+end
+
+@[simp]
+lemma to_jordan_decomposition_to_signed_measure (j : jordan_decomposition α) :
+  (j.to_signed_measure).to_jordan_decomposition = j :=
+(@to_signed_measure_injective _ _ j (j.to_signed_measure).to_jordan_decomposition (by simp)).symm
+
+end jordan_decomposition
+
+namespace signed_measure
+
+open jordan_decomposition
+
+/-- `signed_measure.to_jordan_decomposition` forms a `equiv` between
+`signed_measure α` and `jordan_decomposition α` -/
```

Comment:
```suggestion
/-- `measure_theory.signed_measure.to_jordan_decomposition` and
`measure_theory.jordan_decomposition.to_signed_measure` form a `equiv`. -/
```
Note that any `` `name` `` in a docstring needs to be fully qualified else docgen won't link it.